nativeasset/contract.go

```diff
@@ -102,8 +104,56 @@ func UnpackNativeAssetCallInput(input []byte) (common.Address, common.Hash, *big
 
 // Run implements StatefulPrecompiledContract
 func (c *NativeAssetCall) Run(accessibleState contract.AccessibleState, caller common.Address, addr common.Address, input []byte, suppliedGas uint64, readOnly bool) (ret []byte, remainingGas uint64, err error) {
-	// input: encodePacked(address 20 bytes, assetID 32 bytes, assetAmount 32 bytes, callData variable length bytes)
-	return accessibleState.NativeAssetCall(caller, input, suppliedGas, c.GasCost, readOnly)
+	if suppliedGas < c.GasCost {
+		return nil, 0, vmerrs.ErrOutOfGas
+	}
+	remainingGas = suppliedGas - c.GasCost
```

Comment:
(fyi) I'm thinking of exposing the `Contract.UseGas()` method via `vm.PrecompileEnvironment` to simplify tracking gas. No action required here but this just reminded of it as the change will simplify the `Run()` signature.